pkg/networkservice/ipam/README.md

```diff
@@ -0,0 +1,47 @@
+# Functional requirements
+
+1. For the different test scenarios we need an endpoint, providing IPAM service in at least 2 modes:
+    * point 2 point - allocates pair of IP addresses in /31 subnet;
+    * IP subnet - allocates IP address in some given subnet.
+2. IPAM service is created on list of some IP subnets. Request can set some exclude IP prefixes for the allocated IP
+addresses.
+3. IPAM service should be idempotent, so if we have allocated some IP addresses for the request and request type (p2p,
+subnet) hasn't changed, and allocated addresses are still not excluded by the excluded prefixes, we should return the
+same addresses for the same connection.
+
+# Implementation
+
+## ipamServer
+
+It is a server chain element implementing IPAM service. It works in 2 modes:
+
+### point to point
+
+```go
+conn, _ := ipam.NewServer("10.0.0.0/24").Request(ctx, &networkservice.NetworkServiceRequest{
+    Connection: &networkservice.Connection{
+        Context: &networkservice.ConnectionContext{
+            IpContext: &networkservice.IPContext{
+                DstIpRequired: true, // <-- p2p mode
```

Comment:
I'm not sure that it is the correct way to use `DstIpRequired` flag to select the ipam mode. Probably we can create a separate ipam chain element for subnets. 
@Bolodya1997 , @edwarnicke , @haiodo  Thoughts?

Reply:
I've had more generic question here, do we have an IPAM scenario without SrcIpRequired? Probable switch API to had enum values of required IPs? Thoughts?

Reply:
> I've had more generic question here, do we have an IPAM scenario without SrcIpRequired?

I think no, https://github.com/networkservicemesh/networkservicemesh/search?q=SrcIpRequired in monorepo as I can see no one handles these values.

> Probable switch API to had enum values of required IPs? 

I like idea with enum values  of required IPs